Resources/Prototypes/Catalog/revenant_catalog.yml

```diff
@@ -1,32 +1,51 @@
-#TODO: generic shop system
-- type: revenantListing
-  id: Defile
-  actionId: RevenantDefile
-  listingName: Defile
+- type: listing
+  id: RevenantDefile
+  name: Defile
   description: Costs 30 Essence per use. Defiles the surrounding area, ripping up floors, damaging windows, opening containers, and throwing items. Using it leaves you vulnerable to attacks for a short period of time.
-  icon: Interface/Actions/defile.png
-  price: 10
+  productAction: RevenantDefile
+  cost: 
+    StolenEssence: 10
+  categories:
+  - RevenantAbilities
+  conditions:
+  - !type:ListingLimitedStockCondition
+    stock: 1
 
-- type: revenantListing
-  id: OverloadLights
-  actionId: RevenantOverloadLights
-  listingName: Overload Lights
+- type: listing
+  id: RevenantOverloadLights
+  name: Overload Lights
   description: Costs 40 Essence per use. Overloads all nearby lights, causing the bulbs to shatter and sending out damaging sparks. Using it leaves you vulnerable to attacks for a long period of time.
-  icon: Interface/Actions/overloadlight.png
-  price: 25
+  productAction: RevenantOverloadLights
+  cost: 
+    StolenEssence: 25
+  categories:
+  - RevenantAbilities
+  conditions:
+  - !type:ListingLimitedStockCondition
+    stock: 1
 
-- type: revenantListing
-  id: Blight
-  actionId: RevenantBlight
-  listingName: Blight
+- type: listing
+  id: RevenantBlight
+  name: Blight
   description: Costs 50 Essence per use. Infects all nearby organisms with an infectious disease that causes poison and tiredness. Using it leaves you vulnerable to attacks for a medium period of time.
-  icon: Interface/Actions/blight.png
-  price: 75
+  productAction: RevenantBlight
+  cost: 
+    StolenEssence: 75
+  categories:
+  - RevenantAbilities
+  conditions:
+  - !type:ListingLimitedStockCondition
+    stock: 1
 
-- type: revenantListing
-  id: Malfunction
-  actionId: RevenantMalfunction
-  listingName: Malfunction
+- type: listing
+  id: RevenantMalfunction
+  name: Malfunction
   description: Costs 60 Essence per use. Makes most nearby electronics stop working properly. Using it leaves you vulnerable to attacks for a long period of time.
-  icon: Interface/Actions/malfunction.png
-  price: 125
\ No newline at end of file
+  productAction: RevenantMalfunction
+  cost: 
+    StolenEssence: 125
+  categories:
+  - RevenantAbilities
+  conditions:
+  - !type:ListingLimitedStockCondition
+    stock: 1
```

Comment:
are these intended to have limited stock? it says 'per use' which implies multiple times?

also why does it say 'costs X essence' but then it actually costs a different amount

Reply:
This is a symptom of me being completely incapable of describing things well. The idea is that it's a one-time unlock cost in the menu and then each menu item has an independent use cost each time it's triggered. I keep getting asked about this so i'll remove that portion of it.

Reply:
okay that makes sense